homeassistant/components/utility_meter/sensor.py

```diff
@@ -0,0 +1,241 @@
+"""
+Utility meter from sensors providing raw data.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.utility_meter/
+"""
+import logging
+
+from decimal import Decimal, DecimalException
+import voluptuous as vol
+
+import homeassistant.util.dt as dt_util
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_NAME, ATTR_UNIT_OF_MEASUREMENT,
+    EVENT_HOMEASSISTANT_START, STATE_UNKNOWN, STATE_UNAVAILABLE)
+from homeassistant.core import callback
+from homeassistant.helpers.event import (
+    async_track_state_change, async_track_time_change)
+from homeassistant.helpers.dispatcher import (
+    dispatcher_send, async_dispatcher_connect)
+from homeassistant.helpers.restore_state import RestoreEntity
+from .const import (
+    SIGNAL_START_PAUSE_METER, SIGNAL_RESET_METER,
+    HOURLY, DAILY, WEEKLY, MONTHLY, YEARLY, METER_TYPES)
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_SOURCE_ID = 'source'
+ATTR_STATUS = 'status'
+ATTR_PERIOD = 'meter_period'
+ATTR_LAST_PERIOD = 'last_period'
+ATTR_LAST_RESET = 'last_reset'
+
+CONF_SOURCE_SENSOR = 'source'
+CONF_METER_TYPE = 'cycle'
+CONF_METER_OFFSET = 'offset'
+CONF_PAUSED = 'paused'
+
+ICON = 'mdi:counter'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
```

Comment:
Should we move the config to the component?

Reply:
I made that question to Paulus in discord, but got no answer.

I've kept it here, since that's what other platforms that use configuration file are doing (e.g. MQTT). 

But I have no strong feelings about it.

Reply:
I would suggest moving it.

Reply:
but should it be just a move of the code ? 

or should the configuration of the utility_meter hang in the configuration file beneath the component? if it moves under the component, is it still a sensor?

Reply:
The user should set up the component in the config, not the platform. Then we load sensor platform via discovery. Later config entry is easy, as said above.